03 - CSS Variables/index-START.html

```diff
@@ -1,51 +1,111 @@
-  <title>Scoped CSS Variables and JS</title>
-</head>
-<body>
-  <h2>Update CSS Variables with <span class='hl'>JS</span></h2>
+  <head>
+    <meta charset="UTF-8" />
+    <title>Scoped CSS Variables and JS</title>
+  </head>
+  <body>
+    <h2>Update CSS Variables with <span class="hl">JS</span></h2>
 
-  <div class="controls">
-    <label for="spacing">Spacing:</label>
-    <input id="spacing" type="range" name="spacing" min="10" max="200" value="10" data-sizing="px">
+    <div class="controls">
+      <label for="spacing">Spacing:</label>
+      <input
+        id="spacing"
+        type="range"
+        name="spacing"
+        min="10"
+        max="200"
+        value="10"
+        data-sizing="px"
+      />
 
-    <label for="blur">Blur:</label>
-    <input id="blur" type="range" name="blur" min="0" max="25" value="10" data-sizing="px">
+      <label for="blur">Blur:</label>
+      <input
+        id="blur"
+        type="range"
+        name="blur"
+        min="0"
+        max="25"
+        value="10"
+        data-sizing="px"
+      />
 
-    <label for="base">Base Color</label>
-    <input id="base" type="color" name="base" value="#ffc600">
-  </div>
+      <label for="base">Base Color</label>
+      <input id="base" type="color" name="base" value="#ffc600" />
+    </div>
 
-  <img src="https://source.unsplash.com/7bwQXzbF6KE/800x500">
+    <img src="https://source.unsplash.com/7bwQXzbF6KE/800x500" />
 
-  <style>
+    <style>
+      /* unless specific cases, define CSS variables on the highest component possible to make them accessible to all elements */
 
-    /*
+      /* define default variables needed in root */
+      :root {
+        --base: rgb(182, 142, 142);
+        --spacing: 1rem;
+        --blur: 1rem;
+      }
+
+      /*
       misc styles, nothing to do with CSS variables
-    */
-
-    body {
-      text-align: center;
-      background: #193549;
-      color: white;
-      font-family: 'helvetica neue', sans-serif;
-      font-weight: 100;
-      font-size: 50px;
-    }
-
-    .controls {
-      margin-bottom: 50px;
-    }
-
-    input {
-      width: 100px;
-    }
-  </style>
-
-  <script>
-  </script>
-
-</body>
+      */
+
+      body {
+        text-align: center;
+        background: #000;
+        color: white;
+        font-family: 'helvetica neue', sans-serif;
+        font-weight: 100;
+        font-size: 50px;
+      }
+
+      .controls {
+        margin-bottom: 50px;
+      }
+
+      input {
+        width: 100px;
+      }
+
+      .hl {
+        color: var(--base);
+      }
+
+      img {
+        background: var(--base);
+        padding: var(--spacing);
+        filter: blur(var(--blur));
+      }
+    </style>
+
+    <script>
+      //// selectors
+      const inputs = document.querySelectorAll('.controls input');
+
+      //// callbacks
+      const handleUpdate = (evt) => {
+        const currentInput = evt.target;
+
+        // use the dataset to get the unit of measure (if any)
```

Comment:
G., your code already looks nice and professional. you do not need these kind of comments anymore (or the ones below). maybe try removing the trivial ones and "reflow" the code (remove newlines). can you compare before/after in terms of readability?

Reply:
got your point! shorter code. obvious things removed =  code more readable.